src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedDialog.tsx

```diff
@@ -0,0 +1,176 @@
+import { useState, MouseEvent, FC, useCallback } from 'react';
+import { Button } from 'hds-react/components/Button';
+import { Dialog } from 'hds-react/components/Dialog';
+import { useTranslation } from 'react-i18next';
+
+import { useAppDispatch, useAppSelector } from '@/hooks/common';
+import ProjectProgrammedSearch from './ProjectProgrammedSearch';
+import { IProgrammedProjectSuggestions } from '@/interfaces/searchInterfaces';
+import { IProjectsRequestObject } from '@/interfaces/projectInterfaces';
+import { silentPatchMultipleProjectsThunk } from '@/reducers/projectSlice';
+import { selectSelectedClass, selectSelectedSubClass } from '@/reducers/classSlice';
+import { useOptions } from '@/hooks/useOptions';
+
+interface IDialogProps {
+  handleClose: () => void;
+  isOpen: boolean;
+}
+
+interface IFormState {
+  projectsForSubmit: Array<IProgrammedProjectSuggestions>;
+}
+interface IDialogButtonState {
+  isOpen: boolean;
+}
+
+const DialogContainer: FC<IDialogProps> = ({ isOpen, handleClose }) => {
+  const [formState, setFormState] = useState<IFormState>({
```

Comment:
Should be refactored to:

`const [projectsForSubmit, setProjectsForSubmit] = useState([]);`